lib/travis/api/v3/services/cron/create.rb

```diff
@@ -0,0 +1,24 @@
+module Travis::API::V3
+  class Services::Cron::Create < Service
+    result_type :cron
+    params :interval, :disable_by_build
+
+    def run!
+      raise LoginRequired unless access_control.logged_in? or access_control.full_access?
+      raise NotFound      unless repository = find(:repository)
+      raise NotFound      unless branch = find(:branch, repository)
+      raise ArgumentError, 'interval must be "daily", "weekly" or "monthly"' unless ["daily", "weekly", "monthly"].include?(params["interval"])
+      access_control.permissions(repository).create_cron!
+
+      if branch.cron
+        access_control.permissions(branch.cron).delete!
+        branch.cron.destroy
+      end
+
+      Models::Cron.create(branch: branch,
+                          interval: params["interval"],
+                          disable_by_build: params["disable_by_build"] ? params["disable_by_build"] : false)
+    end
```

Comment:
I'd love if you could move this into a query method. We try to encapsulate anything AR specific there.